csaf-validation/src/test/kotlin/io/github/csaf/sbom/validation/tests/TestsTest.kt

```diff
@@ -22,30 +22,54 @@ import io.github.csaf.sbom.schema.generated.Csaf
 import io.github.csaf.sbom.validation.ValidationSuccessful
 import io.github.csaf.sbom.validation.assertValidationFailed
 import io.github.csaf.sbom.validation.assertValidationSuccessful
+import io.github.csaf.sbom.validation.generated.Testcases
 import io.github.csaf.sbom.validation.goodCsaf
+import kotlin.io.path.Path
+import kotlin.io.path.readText
+import kotlin.test.AfterTest
 import kotlin.test.Test
 import kotlin.test.assertEquals
+import kotlin.test.assertNotNull
+import kotlin.test.assertTrue
+import kotlinx.serialization.json.Json
 
 /** The path to the test folder for the CSAF 2.0 tests. */
 var testFolder: String = "../csaf/csaf_2.0/test/validator/data/"
 
-/**
- * Short utility function to construct the path to the test file based on the test file ID for
- * mandatory tests.
- */
-fun mandatoryTest(id: String): String {
-    return "$testFolder/mandatory/oasis_csaf_tc-csaf_2_0-2021-${id}.json"
-}
+class TestsTest {
 
-/**
- * Short utility function to construct the path to the test file based on the test file ID for
- * optional tests.
- */
-fun optionalTest(id: String): String {
-    return "$testFolder/optional/oasis_csaf_tc-csaf_2_0-2021-${id}.json"
-}
+    val executedTests = mutableSetOf<String>()
+
+    companion object {
+        val testCases =
+            Json { ignoreUnknownKeys = true }
```

Comment:
Why do we need to specify that?

Reply:
Because the file has the `  "$schema": "https://raw.githubusercontent.com/oasis-tcs/csaf/master/csaf_2.0/test/validator/testcases_json_schema.json"` line and I got an error that this property is unknown. I don't know if this is a bug in Kotlinx serialization or if this really "violates" the schema (which would be a bit weird, since it specifies the schema)

Reply:
Technically, yes. This violates the schema. This should be resolved shortly: https://github.com/oasis-tcs/csaf/issues/824

Reply:
Ok, in order to not block process: We will remove the ignore unknown keys as soon as this is resolved in the TC repo.